### ISSUE TITLE
Batch upsert collection fails with "error parsing 'data'"

### ISSUE THREAD

**Issue by rafiqm** [GetStream/stream-php#78]
## Expected Behavior
Example was copied directly from [official documentation](https://getstream.io/docs/php/#collections_batch)

Performing upsert() with an array of arrays should batch insert/update many collection items.

## Actual Behavior
Exception is thrown:
```
Type:        GetStream\Stream\StreamFeedException
Message:     {"detail":"Error parsing 'data'","status_code":400,"code":4,"except
ion":"InputException","exception_fields":{"data":["Error parsing 'data'"]},"dura
tion":"0.16ms"}

```

## Steps to Reproduce the Problem

  1. Install and configure the stream-php library as described in documentation Quickstart
  2. Paste example code from Batch Endpoint documentation
```
$client->collections()->upsert('visitor', [
    [
        'id' => '123',
        'username' => 'johndoe',
        'favorite_color' => 'blue',
    ]
]);
```
 3. Execute code and exception is thrown.

## Troubleshooting

- Tried JS example code which works successfully (ruling out account-specific issues)
- JS lib posts to "collections/" endpoint and PHP lib to "meta/" (Collections.php line 178)
- PHP lib encapsulates "data" in an array resulting in batch requests having an "extra" array compared to the JS code.
- Removing the extra array causes non-batch requests to fail.

## Solution 

I did not create a pull request because this solution is not something I've put a lot of thought into regarding performance or compatibility, it's just good enough to work.

```
public function upsert($collectionName, array $data)
    {
        $shouldBatch = array_filter($data, 'is_array') === $data;

        $response = $this->doRequest('POST', 'collections/', ['data' => [$collectionName => $shouldBatch ? $data : array($data)]]);
        $body = $response->getBody()->getContents();
        return json_decode($body, true);
    }

```

## Specifications

  - Version: 3.0.0
  - Platform: PHP 7.0.33, Linux, Nginx

**Comment by logan-jobzmall** [comment]
Same issue here

**Comment by tbarbugli** [comment]
@pterk sounds like we serialize data incorrectly here, can you have a quick look?

**Comment by pterk** [comment]
@logan-jobzmall I've released a version 3.0.1 with a more flexible upsert method.
